src/sentry/rules/conditions/event_attribute.py

```diff
@@ -269,10 +137,17 @@ def _passes(self, attribute_values: Sequence[object | None]) -> bool:
 
     def passes(self, event: GroupEvent, state: EventState, **kwargs: Any) -> bool:
         attr = self.get_option("attribute", "")
-        try:
-            attribute_values = self._get_attribute_values(event, attr.lower())
-        except KeyError:
+        path = attr.split(".")
+
+        first_attr = path[0]
+        attr_handler = attribute_registry.get(first_attr)
+        if not attr_handler:
             attribute_values = []
+        else:
+            try:
+                attribute_values = attr_handler.handle(path, event)
+            except KeyError:
```

Comment:
🎉 nice that this refactor separates this error handling vs before - it's a lot simpler to reason about the key error.

for extra credit, i think it might be even nicer for us to update the AttributeHandler implementations so each individual handler works as intended (either returning [] or the correct value)

Reply:
i'll add that and still have the KeyError, but capture the exception since we know it actually shouldn't happen